File: php/features.php
<?php
declare(strict_types=1);

// KekkaiCurrency SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KekkaiCurrencyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KekkaiCurrencyBaseFeature();
            case "test":
                return new KekkaiCurrencyTestFeature();
            default:
                return new KekkaiCurrencyBaseFeature();
        }
    }
}

File: py/features.py
# KekkaiCurrency SDK feature factory

from feature.base_feature import KekkaiCurrencyBaseFeature
from feature.test_feature import KekkaiCurrencyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KekkaiCurrencyBaseFeature(),
        "test": lambda: KekkaiCurrencyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
